### ISSUE TITLE
genind2hierfstat is no longer exported and no longer needed

### ISSUE THREAD

**Issue by hlapp** [NESCent/popgenInfo#78]
The function genind2hierfstat has been removed from `adegenet`, as `hierfstat` now deals natively with `genind` objects: http://lists.r-forge.r-project.org/pipermail/adegenet-forum/2015-September/001285.html

This results in the following type of errors:
```R
Quitting from lines 80-87 (DifferentiationSNP.Rmd) 
Error in eval(expr, envir, enclos) : 
  could not find function "genind2hierfstat"
```

**Comment by smanel** [comment]
I try to use genind objects directly with hierfstat in the vignette. It does not work in the case of the commands used in the vignette. 
I am waiting an answer from J. Goudet.

**Comment by bjcochrane** [comment]
genind2hierfstat is indeed gone from adegenet, however hierfstat version 0.4-14, the most recent on CRAN, does not read genind objects (at least for the basic.stats function).  Is there a developer version somewhere that does?

**Comment by zkamvar** [comment]
Hi @bjcochrane, @hlapp, and @smanel 

Sorry for the late reply, I've been buried in work and then a much needed holiday. `genind2hierfstat()` now exists on [the development version of hierfstat](http://github.com/jgx65/hierfstat), which you can install using devtools:

```r
devtools::install_github("jgx65/hierfstat")
```

hope that helps.

**Comment by hlapp** [comment]
So @zkamvar, was the resolution that `genind2hierfstat()` does need to be exported from `hierfstat`? I.e., it did exist in the package since July, it just wasn't exported, with the reasoning that it shouldn't be needed because users shouldn't have to worry about interoperability. I think the functions that are used in the vignette accept `genind` objects and convert automatically (as they should) - was that deemed not desirable to implement for those functions that apparently haven't done so yet?

I'd like to avoid having to mix CRAN and development packages in the container that renders the website (and that thus runs the tests). That's because most users who would be a target audience for the vignettes aren't likely to mix at will either. But we can of course discuss.

**Comment by zkamvar** [comment]
@hlapp, I believe the original idea was to port `genind2hierfstat()` to hierfstat so that genind objects would be passed to `genind2hierfstat()` internally. [It was placed in hierfstat on June 22nd](https://github.com/jgx65/hierfstat/commit/b63d8dccf6a2907b9fa17ae586e1e735fb863b52) as an internal function called `.genind2hierfstat()`. However, hierfstat has not been updated on CRAN in over a year and it only recently gained the ability to work with genind objects for many of the functions since [this commit six days ago](https://github.com/jgx65/hierfstat/commit/2056547e55bf39f9abc7c8b8b76796a3f5adbd8d). 

Additionally, the author has stated that [hierfstat will be on CRAN by the end of the year](https://github.com/jgx65/hierfstat/issues/8#issuecomment-154436340) and has [asked for people to check their data with the functions](https://github.com/jgx65/hierfstat/issues/9).

I agree that we want to avoid mixing CRAN and github packages, but until these are submitted to CRAN, there is very little we can do. Is it possible to have a development version of the website?

**Comment by jgx65** [comment]
Hi all, I'm currently writing a short vignette explaining the latest developments in `hierfstat`. As soon as this is done, I'll put it on CRAN.   Hope this is ok for you?

**Comment by jgx65** [comment]
Currently, we cannot avoid completely using `hierfstat::genind2hierfstat` because some functions (e.g. `test.g`, `varcomp.glob` ) require two data frames, one for loci and one for populations or levels. Several others (`basic.stats`, `wc`, `genet.dist`, `ind.pca`,...) work with genind objects.

**Comment by hlapp** [comment]
I updated the Docker file underlying the image that Circle CI uses, and the [Docker image is now rebuilt](https://hub.docker.com/r/hlapp/rpopgen/builds/blteoiszrzg9brqgtqvug9c/).

When rebuilding with that new image, the [original error](https://github.com/NESCent/popgenInfo/issues/78#issue-117830254) disappears, but [a new one appears](https://github.com/NESCent/popgenInfo/pull/79#issuecomment-160983793).  @smanel had already [committed the fix](08cf0f1c96f739446ccd4fb668b8e5c9d8181868) to `DifferentiationSNP.md`, which I subsequently moved from `master` to a branch, from which I now created a pull request (#79) so we can track progress and build status there. Once we get that to a passing state, we can merge into `master`, which will hopefully make that build pass again, too, and I'll then pull the changes into the `microsatellites` branch on which @smanel has been working on a vignette.

**Comment by jgx65** [comment]
@hlapp @smanel @zkamvar Latest version of hierfstat (0.04.20) submitted to CRAN.

**Comment by zkamvar** [comment]
:+1: :confetti_ball: :tada: :100:

**Comment by jgx65** [comment]
It's now that I'm gonna get flamed ;-)

**Comment by hlapp** [comment]
This has been fixed through #79 and #87.
